src/Compilers/Core/Portable/InternalUtilities/ISetExtensions.cs

```diff
@@ -31,6 +32,17 @@ public static bool AddAll<T>(this ISet<T> set, ImmutableArray<T> values)
             return result;
         }
 
+        public static bool AddAll<T, S>(this ISet<T> set, ImmutableArray<S> values, Func<S, T> selector)
```

Comment:
>AddAll [](http://example.com/codeflow?start=27&length=6)

It looks like the new `AddAll` and `RemoveAll` overloads are only used by `getSynthesizedClosureMethods`, and the behavior seems uncommon with a `selector`. Given that, and that we're not using the return values in that caller, consider simply inlining the loops in that one caller.